arrow-fx-coroutines-stream/src/main/kotlin/arrow/fx/coroutines/stream/Chunk.kt

```diff
@@ -973,9 +973,9 @@ abstract class Chunk<out O> {
               val next = rem.first()
               val nextSize = next.size()
               when {
-                nextSize < toDrop -> loop(rem.tail(), toDrop - nextSize)
-                nextSize == toDrop -> Queue(rem.tail(), size - n)
-                else -> Queue(next.drop(toDrop) prependTo rem.tail(), size - n)
+                nextSize < toDrop -> loop(rem.tailOrNull()!!, toDrop - nextSize)
```

Comment:
I think the exception thrown by `tail` would be more descriptive than a NPE here, no? Otherwise I believe you could remove the `tail` function as it's not used anymore

Reply:
`tail` is indeed not used anymore, just had to double check. We could move it as an extension method to the `Stream` module instead and keep the more descriptive error message.

Reply:
Moved `tail` to Streams, and gave a more descriptive message.